File: array/665.cpp
class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int n = nums.size();
        if(n <= 1)return 1;
        int cnt = 0;
        for(int i = 1; i < n && cnt < 2; ++i)
            if(nums[i - 1] > nums[i]){
                ++cnt;
                if(i == 1 || nums[i - 2] <= nums[i])
                    nums[i - 1] = nums[i];
                else
                    nums[i] = nums[i - 1];
            }
        return cnt < 2;
    }
};

File: array/665.java
class Solution {
    public boolean checkPossibility(int[] nums) {
        int n = nums.length;
        if(n == 0)return true;
        boolean flag = false;
        for(int i = 0, pre = 0; i < n - 1; pre = nums[i], ++i)
            if(nums[i] > nums[i + 1]){
                if(!flag){
                    if(i == 0 || pre <= nums[i + 1])
                        nums[i] = nums[i + 1];
                    else
                        nums[i + 1] = nums[i];
                    flag = true;
                }
                else
                    return false;
            }
        return true;
    }
}
